pkg/envoy/ads/response.go

```diff
@@ -122,15 +125,31 @@ func (s *Server) SendDiscoveryResponse(proxy *envoy.Proxy, request *xds_discover
 	}
 
 	resourcesSent := mapset.NewSet()
+	subscribedResources := proxy.GetSubscribedResources(typeURI)
 	for _, res := range resourcesToSend {
 		proto, err := ptypes.MarshalAny(res)
 		if err != nil {
 			log.Error().Err(err).Str(errcode.Kind, errcode.ErrMarshallingXDSResource.String()).
 				Msgf("Error marshalling resource %s for proxy %s", typeURI, proxy.GetCertificateSerialNumber())
 			continue
 		}
+		// Add resource to response
 		response.Resources = append(response.Resources, proto)
-		resourcesSent.Add(cache.GetResourceName(res))
+
+		// We will preemptively filter each of the resources generated by the non-wildcard verticals and NOT count them
+		// towards `LastSentResources` if they are not subscribed resources if they were never requested by the proxy.
```

Comment:
Could you reword this statement with a double `if` for clarity

Reply:
will do

Reply:
I hope I understood you right here, lmk if not